Sources/ProjectSpec/SpecValidation.swift

```diff
@@ -154,38 +154,17 @@ extension Project {
         }
 
         for target in targets {
+            var uniqueDependencies = Set<Dependency>()
+
             for dependency in target.dependencies {
-                switch dependency.type {
-                case .target:
-                    let dependencyTargetReference = try TargetReference(dependency.reference)
-
-                    switch dependencyTargetReference.location {
-                    case .local:
-                        if getProjectTarget(dependency.reference) == nil {
-                            errors.append(.invalidTargetDependency(target: target.name, dependency: dependency.reference))
-                        }
-                    case .project(let dependencyProjectName):
-                        if getProjectReference(dependencyProjectName) == nil {
-                            errors.append(.invalidTargetDependency(target: target.name, dependency: dependency.reference))
-                        }
-                    }
-                case .sdk:
-                    let path = Path(dependency.reference)
-                    if !dependency.reference.contains("/") {
-                        switch path.extension {
-                        case "framework"?,
-                             "tbd"?,
-                             "dylib"?:
-                            break
-                        default:
-                            errors.append(.invalidSDKDependency(target: target.name, dependency: dependency.reference))
-                        }
-                    }
-                case .package:
-                    if packages[dependency.reference] == nil {
-                        errors.append(.invalidSwiftPackage(name: dependency.reference, target: target.name))
-                    }
-                default: break
+                let dependencyValidationErrors = try validate(dependency, in: target)
+                errors.append(contentsOf: dependencyValidationErrors)
+
+                if uniqueDependencies.contains(dependency) {
+                    errors.append(.duplicateDependencies(target: target.name, dependencyReference: dependency.reference))
+                    continue
```

Comment:
If a target has more than one duplicated dependency, this will bail out after the first one and not show all the errors. It will also prevent any further dependency validation errors from appearing. Do we need to `continue` here?

Reply:
That is an inner continue statement, it will just continue to the next dependency, not the next target
It's just, but actually I can delete it, because I use if-else and there won't be extra insert to set attempt anyway

Reply:
Removed the `continue` statement, because it only introduced ambiguity
And explicitly added cases which you mentioned to `ProjectSpecTests`